packages/forms/backend/sirius-components-widget-reference/src/main/java/org/eclipse/sirius/components/widget/reference/browser/ModelBrowserDescriptionProvider.java

```diff
@@ -240,10 +245,37 @@ private boolean hasChildren(VariableManager variableManager) {
             hasChildren = !resource.getContents().isEmpty();
         } else if (self instanceof EObject eObject) {
             hasChildren = !eObject.eContents().isEmpty();
+            hasChildren = hasChildren && this.hasCompatibleDescendants(variableManager, eObject, this.resolveReference(variableManager).orElse(null));
+        }
+        return hasChildren;
+    }
+
+    private boolean hasChildren(VariableManager variableManager, EReference reference) {
+        Object self = variableManager.getVariables().get(VariableManager.SELF);
+
+        boolean hasChildren = false;
+        if (self instanceof Resource resource) {
+            hasChildren = !resource.getContents().isEmpty();
+        } else if (self instanceof EObject eObject) {
+            hasChildren = !eObject.eContents().isEmpty();
+            hasChildren = hasChildren && this.hasCompatibleDescendants(variableManager, eObject, reference);
         }
         return hasChildren;
     }
 
+    private boolean hasCompatibleDescendants(VariableManager variableManager, EObject eObject, EReference reference) {
+        var iter = eObject.eAllContents();
```

Comment:
I'm not confortable with the `eAllContents` here.
Could you try something recursive with `eContents` instead?

Reply:
Yeah, I missed it.
I used `eContents` and a recursive methode,
I 'll add some TU for this class, the algorithm is quite complex.

Reply:
Tests will be contributed in a separate commit.